pkg/yang/entry.go

```diff
@@ -504,17 +509,27 @@ func ToEntry(n Node) (e *Entry) {
 			// available.  There is nothing else for us to do.
 		case "include":
 			for _, a := range fv.Interface().([]*Include) {
-				// Specifically handle the case that the submodule's name is the same
-				// as the node that we are currently converting to an entry. This allows
-				// us to ensure that we do not process circular dependencies that exist
-				// in some module's include chains.
+				// Handle circular dependencies between submodules. This can occur in
+				// two ways:
+				//  - Where submodule A imports submodule B, and vice versa then the
+				//    whilst processing A we will also try and process A (learnt via
+				//    B). The default case of the switch handles this case.
+				//  - Where submodule A imports submodule B that imports C, which also
+				//    imports A, then we need to check whether we already have merged
+				//    the specified module during this parse attempt. We check this
+				//    against a map of merged submodules.
+				_, merged := mergedSubmodule[a.Module.NName()][n.NName()]
```

Comment:
Maybe you can simplify this to a single map.  Two way I can think of:

key := a.Module.NName() + ":" + n.NName()

key := struct {
    Module string
    Include string
} { Module: a.Module.NName(), Include: n.NName()}

Obviously the second case should use a names type.

Now you can just say

merged := mergedSubmodule[key] 

If you want to use

_, merged := mergedSubmodule[key] 

then mergedSubmodule should be a map to struct{}